### PR TITLE
TH2Poly Tracker Maps

### PR DESCRIPTION
**Pull request by alaperto**
#### PR description:

In order to exploit jsroot with interactive tracker maps for both Strip and Pixel, need to save plots (in form of TH2Poly) inside .root files.
Changes:
- Strip: introduced TH2Poly in new canvas, saved to .root
- Pixel: already in TH2Poly format, changed bin name and saved new canvas to .root

New TH2Poly plots are being integrated into the Tracker Map website (outside CMSSW) with a [separate PR](https://github.com/CMSTrackerDPG/cmsTrackerMaps/pull/5).

#### PR validation:

1. Old format (png files) used in previous tracker map website version are not modified
2. New TH2Poly plots inside root files are as expected, examples:
- Strip: https://alaperto.web.cern.ch/htmls/strip_jsroot.html
- Pixel: https://alaperto.web.cern.ch/htmls/pixel_jsroot.html
3. [Preview of the website](https://alaperto.web.cern.ch/tkmaps/index.html?link=true&mode=compare&checkbox88=true&checkbox104=true&refRunPath=&currRunPath=&mapSelect=undefined) with interactive Strip and Pixel Maps.

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * addpkg = Alignment/OfflineValidation,CondCore/SiStripPlugins,CondCore/AlignmentPlugins,CondTools/SiStrip

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
failure are unrelated. The IB is broken...

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by emanueleusai** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Comment by rappoccio** [comment]
+1

File: CommonTools/TrackerMap/src/TrackerMap.cc [hunk 1]
```diff
@@ -23,6 +23,9 @@
 #include "TArrow.h"
 #include "TLegend.h"
 #include "TH1F.h"
+#include "TH2Poly.h"
+#include "TGraph.h"
+#include "TPaletteAxis.h"
```
Comment (makortel) [inline CommonTools/TrackerMap/src/TrackerMap.cc:28]: Just to note that this dependence would have required the addition of `roothistpainter` dependence in the `BuildFile.xml` to keep UBSAN build functional. I have a fix in https://github.com/cms-sw/cmssw/pull/42095.

**Pull request merged**